augur/augur_client.py

```diff
@@ -322,33 +303,24 @@ def _contracts_from_abi(self, abi):
         contracts[name] = contract
     return contracts
 
-  async def _start_new_block_listener(self):
-    self._require_is_open()
-    ethereum_uri = self._ethereum_client.providers[0].endpoint_uri
-    websocket = await websockets.connect(ethereum_uri)
-    await self._send_request('eth_subscribe', websocket, ['newHeads'])
-    await asyncio.sleep(0)
-    async for block_event in websocket:
-      block_event_json = json.loads(block_event)['params']['result']
-      self._latest_block = int(block_event_json['number'], 16)
-
-  async def _fetch_historical_events(self, event_name_list, current_block,
-      on_event, increment_block=1000):
-    topic_list = []
-    for event_name in event_name_list:
-      event_map = self._event_name_to_signature_map.get(event_name, None)
-      if event_map:
-        topic_list.append(event_map)
-    while current_block < self._latest_block:
-      await asyncio.sleep(0)
-      event_filter = self._ethereum_client.eth.filter({
-        'address': self._ethereum_client.toChecksumAddress(
-          self._addresses['Augur']),
-        'topics': [topic_list],
-        'fromBlock': current_block,
-        'toBlock': current_block + increment_block
-      })
-      events = event_filter.get_all_entries()
+  async def _filter_blocks(self, start_block, end_block, filter, handler,
+      increment):
+    filter['address'] = self._ethereum_client.toChecksumAddress(
+      self._addresses['Augur'])
+    for current_block in range(start_block, end_block, increment):
+      filter['fromBlock'] = current_block
+      filter['toBlock'] = current_block + increment
+      events_filter = self._ethereum_client.eth.filter(filter)
+      events = events_filter.get_all_entries()
       for event in events:
-        on_event(event)
-      current_block += increment_block
+        await handler(event)
```

Comment:
It will schedule it for immediate execution. We already have `filter_blocks` scheduled for execution later on. Do we want to schedule handler to run at a later time as well? In the last commit, I removed async from the handler method, so handler calls will block the event loop, which is probably what we don't want.

Reply:
Handler calls should not block the event loop.